src/views/Home.jsx

```diff
@@ -1,14 +1,57 @@
 import './Home.css';
+import { useState, useCallback } from 'react';
+import { getItemData, streamListItems } from '../api';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+
+export function Home({ setListToken }) {
+	const [tokenInput, setTokenInput] = useState('');
+
+	const handleClick = useCallback(() => {
+		const newToken = generateToken();
+		setListToken(newToken);
+	}, [setListToken]);
+
+	const tokenHandler = (e) => {
+		setTokenInput(e.target.value);
+	};
+
+	const handleTokenSubmit = (e) => {
+		e.preventDefault();
+		streamListItems(tokenInput, (snapshot) => {
```

Comment:
I fully admit to not being a Firebase/Firestore expert, but I suspect there's a better method we could use for this. My understanding is that `streamListItems` opens a stream that listens for updates. Since we really just want to check if a list (`collection`) exists, I suspect there would be a better function we could use to just check the database once versus opening a stream/snapShot of the data. When I went digging in the documentation, I found this https://firebase.google.com/docs/firestore/query-data/get-data#example_data, which could be used to access a collection. I printed out what came back and saw that there is a `size` property. My gut says this would be a better way to test if a list exists (if you go this route, I would add/export a new function in `firebase.js`). Let me know if you have any questions about this!

Reply:
Thank you, Lauren! Now, we are importing a function from firebase.js, and we are using getDocs and .empty to check.